src/hotspot/share/services/memReporter.hpp

```diff
@@ -83,10 +83,17 @@ class MemReporterBase : public StackObj {
 
     size_t amount = s1 - s2;
     assert(amount <= SIZE_MAX - _scale / 2, "size_t overflow");
```

Comment:
The assert `assert(amount <= SIZE_MAX - _scale / 2, "size_t overflow");` is not needed.